pkg/agent/multicluster/pod_route_controller.go

```diff
@@ -0,0 +1,433 @@
+	gwInformer cache.SharedIndexInformer,
+	client openflow.Client,
+	ovsBridgeClient ovsconfig.OVSBridgeClient,
+	interfaceStore interfacestore.InterfaceStore,
+	nodeConfig *config.NodeConfig,
+	namespace string,
+) *MCPodRouteController {
+	controller := &MCPodRouteController{
+		k8sClient:       k8sClient,
+		mcClient:        mcClient,
+		ovsBridgeClient: ovsBridgeClient,
+		ofClient:        client,
+		interfaceStore:  interfaceStore,
+		nodeConfig:      nodeConfig,
+		podQueue:        workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "MCPodRouteControllerForPod"),
+		gwQueue:         workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "MCPodRouteControllerForGateway"),
+		gwInformer:      gwInformer,
+		gwLister:        mclisters.NewGatewayLister(gwInformer.GetIndexer()),
+		namespace:       namespace,
+		podCache:        make(map[string]*podNodeIPs),
+		ipToPods:        make(map[string]sets.String),
+		podWorkerStopCh: make(chan struct{}),
+	}
+
+	controller.gwInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(cur interface{}) {
+				controller.enqueueGateway(cur)
+			},
+			// Gateway UPDATE event doesn't impact Pod flows, so ignore it.
+			DeleteFunc: func(old interface{}) {
+				controller.enqueueGateway(old)
+			},
+		},
+		resyncPeriod,
+	)
+	return controller
+}
+
+func (c *MCPodRouteController) createPodInformer() {
+	listOptions := func(options *metav1.ListOptions) {
+		options.FieldSelector = fields.OneTermNotEqualSelector("spec.nodeName", c.nodeConfig.Name).String()
+	}
+	c.podInformer = coreinformers.NewFilteredPodInformer(
+		c.k8sClient,
+		metav1.NamespaceAll,
+		0,
+		nil,
+		listOptions,
+	)
+	c.podInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(cur interface{}) {
+				c.enqueuePod(cur, false)
+			},
+			UpdateFunc: func(old, cur interface{}) {
+				c.enqueuePod(cur, false)
+			},
+			DeleteFunc: func(old interface{}) {
+				c.enqueuePod(old, true)
+			},
+		},
+		resyncPeriod,
+	)
+	c.podLister = corelisters.NewPodLister(c.podInformer.GetIndexer())
+}
+
+func (c *MCPodRouteController) enqueueGateway(obj interface{}) {
+	_, isGW := obj.(*mcv1alpha1.Gateway)
+	if !isGW {
+		klog.ErrorS(errors.New("received unexpected object"), "enqueueGateway can't process event", "obj", obj)
+		return
+	}
+	c.gwQueue.Add(dummyKey)
+}
+
+// validatePod validates the Pod info to determine if it's a valid Pod which should
+// have	Openflow flows installed for cross-cluster traffic.
+func (c *MCPodRouteController) validatePod(pod *corev1.Pod) bool {
+	// pod.Status.PodIP and HostIP should be an IPv4 address for IPv4 and dual stack clusters.
+	// IPv6 only cluster is not supported by Antrea Multi-cluster yet.
+	podIP := pod.Status.PodIP
+	podNodeIP := pod.Status.HostIP
+
+	if pod.Spec.NodeName == c.nodeConfig.Name {
```

Comment:
This should not happen as you added a filter to Informer.

Reply:
Yes, removed this check.